src/components/Checkbox/Checkbox.tsx

```diff
@@ -11,8 +11,10 @@ export interface CheckboxProps {
     checkboxId?: string;
     /** ClassName that appears in addition to "checkbox" */
     className?: string;
-    /* The current selected state of the checkbox */
-    isSelected?: boolean;
+    /* For a controlled component only. If checked is not defined, the state of the uncontrolled native `input` component will be used. */
+    checked?: boolean;
+    /* For uncontrolled component only. The initial value of an uncontrolled checkbox component. */
```

Comment:
ah I see, this is gonna be a breaking change.  Let's keep this, but put in the comments here (and in the stories) that we should be using the controlled version, and that `defaultChecked` will be deprecated in the next major release.

Reply:
You mentioned it's a best practice to use controlled components only and that's mostly true. The React team does recommend using controlled components but they also recognized that when using form elements with other libraries, you may need to use uncontrolled components and `ref`. This is the use case I ran into when using `react-hook-form`. For that library, a `ref` is passed to subscribe to the library to keep track of data. There are alternatives for this and I can change the `Checkbox` component to be controlled, but this would be a bigger manual change.

I would keep both versions and not deprecate the usage of uncontrolled components for those cases where it's useful.

Reply:
Yeah we want to allow either `checked` (controlled) or `defaultChecked` (uncontrolled) props, as there are use cases for both. The breaking change is removing the `isSelected` prop.

Reply:
Thanks for the clarification - let's keep both then.  I know it's generally the responsibility of the app developer to know stuff that's written in React documentation, but it would be helpful to have a few lines in the comments that point a new developer to which version they should be using, in case they're confused.